### DISCUSSION TITLE
Generating from a local copy of a schema with HTTP references

### DISCUSSION THREAD

**Discussion by pjohnmeyer** [wol-soft/php-json-schema-model-generator#61] [Q&A]
Today I tried running the generator on a copy of a schema downloaded from a private repository. Every schema file in the schema has a fully-qualified `https` id of the form `https://example.com/schema/...`, and thus uses those URIs for `$ref` attributes. However, I need to generate models against the local copy of the schema, not the files that reside at that URL.

My expectation was that if I loaded all of the local schema files using a `RecursiveDirectoryProvider`, this library would note those schemas were already loaded and not attempt to fetch them over HTTPS. However, I could not find a way to make this happen, and it always attempts to load references over HTTPS.

```php
use PHPModelGenerator\ModelGenerator as Generator;
use PHPModelGenerator\SchemaProvider\RecursiveDirectoryProvider

(new Generator())->generateModels(
    new RecursiveDirectoryProvider(__DIR__ . '/path/to/schema'),
    __DIR__ . '/path/to/output'
);
```

Is there some configuration I am missing to go into offline mode, or perhaps a different way to load the schemas? Or would this be a new feature for the library.

Thanks!

**Comment by wol-soft** [comment]
Hi,

can you provide an example schema? Generally the `$ref` resolves to the value which is provided. Consequently a provided URI will resolve to the URI. If I understand your concern correctly, you download the files separately (into the same directory I guess?). Then the `$ref` must contain the file to be able to determine which file contains the referenced schema (for example: `"$ref":"referencedSchema.json#https://example.com/schema/referencedSchema.json"`).

**Reply by pjohnmeyer** [reply]
I'll have to construct an artificial example which I may not get to for a few days, as the schema I am referring to is currently private, but yes I can provide a more concrete example of what I'm trying to do.
